hs_app_timeseries/forms.py

```diff
@@ -318,28 +574,97 @@ def _get_cv_dropdown_widget_items(dropdown_items, selected_item_name):
     # create a list of tuples
     cv_items = [(item.name, item.name) for item in dropdown_items]
 
-    # add the selected item as a tuple to the beginning of the list of items
-    # so that it will be displayed as the currently selected item
-    cv_items = [(selected_item_name, selected_item_name)] + cv_items
+    # alphabetize the terms
+    cv_items.sort(key=lambda tup: tup[0])
+
+    if selected_item_name is None or len(selected_item_name) == 0:
+        selected_item_name = NO_SELECTION_DROPDOWN_OPTION
+        cv_items = [(selected_item_name, selected_item_name)] + cv_items
+    else:
+        cv_items = [(selected_item_name, selected_item_name)] + cv_items + \
+                   [(NO_SELECTION_DROPDOWN_OPTION, NO_SELECTION_DROPDOWN_OPTION)]
+
     cv_item_choices = tuple(cv_items)
     return cv_item_choices
 
+
+def _set_available_elements_form_field(form, elements, element_name):
+    elements_data = []
+    for element in elements:
+        element_data = model_to_dict(element, exclude=["object_id", "series_ids", "content_type"])
+        elements_data.append(element_data)
+    element_data_str = ""
+    if len(elements_data) > 0:
+        element_data_str = json.dumps(elements_data)
+    form_field_name = "available_{}s".format(element_name)
```

Comment:
perhaps ```s``` needs to be removed since it may not apply to the plural of some element_name, e.g., some element_name may require ```es```. To avoid the grammar complexity, perhaps simply removing ```s``` is sufficient?